### ISSUE TITLE
Pressing tab in InputTextMultiline adds two tabs

### ISSUE THREAD

**Issue by ongamex** [ocornut/imgui#1336]
Here is my test code:

```
static char code[1024 * 1024];
ImGui::InputTextMultiline("##source", code, SGE_ARRSZ(code), ImVec2(0,0), ImGuiInputTextFlags_AllowTabInput);
```

The character is handled twice, and as far as I know this code isn't needed anymore:
```
else if ((flags & ImGuiInputTextFlags_AllowTabInput) && IsKeyPressedMap(ImGuiKey_Tab) && !io.KeyCtrl && !io.KeyShift && !io.KeyAlt && is_editable) { .... }
```

**Comment by ocornut** [comment]
Hello,
I cannot repro the bug here.
I think the reason is that your backend emits both a `ImGuiKey_Tab` keypress (sets a key within `io.KeyDown[]` array) and a `\t` character input with `io.AddInputCharacter()`. Which backend are you using?

**Comment by ongamex** [comment]
Hmm, this is probably the issue, I'll give it a try in a few hours. Thanks!

PS.
I wrote my own win32/x11 backend.

**Comment by ongamex** [comment]
Yep, that was the case, thanks!

**Comment by ocornut** [comment]
Taking note. I intend to add an applet that is a sort of "checklist" of imgui integration and that's one interesting case to check.

**Comment by ocornut** [comment]
Hello @ongamex ,
As it turns out, the problem was maybe elsewhere, see the comments in #2467!

**Comment by ocornut** [comment]
This broke recently (on some backend, mostly Win32, SFML, Allegro + trickling input queue).
Fixed by dfbe938e54858b8b443ca2981d037cc2f35cc931
